authentication/views.py

```diff
@@ -4,20 +4,32 @@
 from authentication.permissions import CustomUserUpdatePermission
 from authentication.serializers import (UserRegistrationSerializer, UserUpdateSerializer, UserPasswordUpdateSerializer)
 from django.contrib.auth import authenticate
-from django.conf import settings
-from django.http import JsonResponse
-from rest_framework import status, generics
-from rest_framework.views import APIView
+from rest_framework.permissions import AllowAny, IsAuthenticated, IsAdminUser
+from rest_framework.exceptions import ValidationError
 from rest_framework.response import Response
-from rest_framework_simplejwt.tokens import RefreshToken
-from rest_framework.permissions import (AllowAny, IsAuthenticated, IsAdminUser)
+from rest_framework.views import APIView
+from authentication.models import CustomUser
+from authentication.permissions import CustomUserUpdatePermission, IsNotAuthenticated
+from authentication.serializers import UserRegistrationSerializer, UserUpdateSerializer, UserPasswordUpdateSerializer
+from authentication.authentications import UserAuthentication
+from forum.errors import Error
 
 
 
-class UserRegistrationView(generics.CreateAPIView):
-    queryset = CustomUser.objects.all()
-    permission_classes = (AllowAny,)
-    serializer_class = UserRegistrationSerializer
+
+
+class UserRegistrationView(APIView):
+
+    authentication_classes = [UserAuthentication]
```

Comment:
One potential issue is the use of a custom ```UserAuthentication``` class, which might not be necessary for a registration view since users typically aren't authenticated yet when registering

Reply:
Agree, class removed.